SU2_CFD/src/fluid/CNonPolytropicPengRobinson.cpp

```diff
@@ -0,0 +1,582 @@
+/*!
+ * \file CNonPolytropicPengRobinson.cpp
+ * \brief Source of the non-polytropic Peng-Robinson model.
+ * \author B. Fuentes Monjas
+ * \version 7.3.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../include/fluid/CNonPolytropicPengRobinson.hpp"
+
+CNonPolytropicPengRobinson::CNonPolytropicPengRobinson(su2double gamma, su2double R, su2double Pstar, su2double Tstar, su2double w, const CConfig *config)
+    : CPengRobinson (gamma, R, Pstar, Tstar, w) {
+  Temperature = TstarCrit; // Initialization of Temperature to critical value
+}
+
+void CNonPolytropicPengRobinson::SetCpModel(const CConfig* config) {
+  for (int i = 0; i < config->GetnPolyCoeffs(); ++i) {
+    coeffs_[i] = config->GetCp_PolyCoeffND(i);
+  }
+}
+
+su2double CNonPolytropicPengRobinson::T_v_h(su2double v, su2double h){
+  su2double toll = 1e-9, FACTOR = 0.2;
```

Comment:
Would you be able to add more comments in these functions?  A lot is happening here in the code but many variables/code sections don't provide insight to the function.